### PR TITLE
move elasticsearch functionality out of pillows

### PR DESCRIPTION
**Pull request by czue**
should be straightforward to review either way.

@benrudolph cc @snopoke

### CODE REVIEW

**Comment by snopoke** [comment]
:+1: :clock1: :construction:

File: pillowtop/listener.py [hunk 1]
```diff
@@ -456,26 +444,21 @@ def __init__(self, create_index=True, online=True, **kwargs):
...
-        if self.online and (index_exists or create_index):
-            pillow_logging.info("Pillowtop [%s] Initializing mapping in ES" % self.get_name())
-            self.initialize_mapping_if_necessary()
+        if self.online:
+            index_exists = pillow_index_exists(self)
+            if create_index and not index_exists:
+                create_index_for_pillow(self)
+            if create_index or index_exists:
+                pillow_logging.info("Pillowtop [%s] Initializing mapping in ES" % self.get_name())
+                self.initialize_mapping_if_necessary()
```
Comment (benrudolph) [inline pillowtop/listener.py]: how come you decided to leave this in the pillow and took the create_index_for_pillow out of it? also, is there ever a time where we'd want to create the index but now the mapping?

**Comment by benrudolph** [comment]
sweeeet

File: pillowtop/listener.py (hunk 1 above)
Comment (czue) [inline pillowtop/listener.py]: just because i didn't get to it yet. had a meeting and thought i'd sneak in a checkpoint while the :construction: ran

Comment (benrudolph) [inline pillowtop/listener.py]: :+1:

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**